File: frontend/src/types/company-profile.ts
export interface CompanyContact {
  website: string;
  phone: string;
  instagram: string;
  tiktok: string;
  youtube: string;
  hashtag: string;
  whatsapp: string;
  facebook: string;
  // Kolom `contact` di DB adalah JSON bebas, dan slot footer template bisa memakai
  // key lain (whatsapp/facebook/location/booking — lihat SocialIcon). Index signature
  // ini membuat contact dapat dibaca per-slot (contact[slot]) tanpa cast.
  [key: string]: string;
}

export interface CompanyProfile {
  id: string;
  user_id: string;
  business_name: string;
  industry: string;
  logo_url: string | null;
  brand_colors: string[] | null;
  brand_font: string | null;
  tagline: string | null;
  /** Alamat bisnis — kolom tersendiri, bukan bagian dari `contact`. Mengisi slot footer
   *  `location` lewat buildFooterContact() saat template dirender branded. */
  address: string | null;
  contact: CompanyContact | null;
  language_preference: string;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/company_profile.py
import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class ContactInfo(BaseModel):
    website: str = ""
    phone: str = ""
    instagram: str = ""
    tiktok: str = ""
    youtube: str = ""
    hashtag: str = ""


class CompanyProfileCreate(BaseModel):
    business_name: str
    industry: str
    logo_url: str | None = None
    brand_colors: list[str] | None = None
    brand_font: str | None = None
    tagline: str | None = None
    contact: ContactInfo | None = None
    language_preference: str = "id"


class CompanyProfileUpdate(BaseModel):
    business_name: str | None = None
    industry: str | None = None
    logo_url: str | None = None
    brand_colors: list[str] | None = None
    brand_font: str | None = None
    tagline: str | None = None
    contact: ContactInfo | None = None
    language_preference: str | None = None


class CompanyProfileData(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    business_name: str
    industry: str
    logo_url: str | None
    brand_colors: list[str] | None
    brand_font: str | None
    tagline: str | None
    contact: dict[str, Any] | None
    language_preference: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
